### ISSUE TITLE
I-de meter reading down or broken?

### ISSUE THREAD

**Issue by alainsch** [ldotlopez/ha-ideenergy#21]
Hello,

Not a problem directly related to this integration, but I don't get any meter reading in HA. 

When I try to use PostMan to do the API calls myself, I get an error: "Se ha producido un error. No se puede completar la operación solicitada."

When I check the i-de site with my account, I see no readings after 7th of july and with "monitor de consumo" > "Consult potencia en tiempo real" I get an "Error de conexión".

Anybody else having this problem? My account is still an "Usuario Avanzado."

Thanks

Alain

**Comment by ad-ha** [comment]
Hi,

Mine is working fine. I had similar issues over a month ago, and it got back working alone.

It might be an issue with I-DE on your area or with the meter itself. Give it sometime to see if it comes back online, or either contact I-DE after a couple of weeks.

Cheers

**Comment by arpia49** [comment]
> Hello,
> 
> Not a problem directly related to this integration, but I don't get any meter reading in HA.
> 
> When I try to use PostMan to do the API calls myself, I get an error: "Se ha producido un error. No se puede completar la operación solicitada."
> 
> When I check the i-de site with my account, I see no readings after 7th of july and with "monitor de consumo" > "Consult potencia en tiempo real" I get an "Error de conexión".
> 
> Anybody else having this problem? My account is still an "Usuario Avanzado."
> 
> Thanks
> 
> Alain

Working fine for me.

**Comment by ldotlopez** [comment]
Mine is working too, but I'm running the latest **pre**-release :shrug:

**Comment by francisco-navarro** [comment]
Mine is only working the first time, when the integration login:

2022-07-30 08:12:57 INFO (MainThread) [ideenergy] Login successful for ****
2022-07-30 08:12:57 DEBUG (MainThread) [custom_components.ideenergy.accumulated] restore state: Got *** kWh
2022-07-30 08:17:17 DEBUG (MainThread) [custom_components.ideenergy.accumulated.barrier] Execution denied: update window is closed
2022-07-30 08:17:17 DEBUG (MainThread) [custom_components.ideenergy.accumulated.barrier] Random delay: 4.4 seconds
2022-07-30 08:21:37 DEBUG (MainThread) [custom_components.ideenergy.accumulated.barrier] Execution denied: update window is closed
2022-07-30 08:21:37 DEBUG (MainThread) [custom_components.ideenergy.accumulated.barrier] Random delay: 4.7 seconds
2022-07-30 08:25:57 DEBUG (MainThread) [custom_components.ideenergy.accumulated.barrier] Execution denied: update window is closed
2022-07-30 08:25:57 DEBUG (MainThread) [custom_components.ideenergy.accumulated.barrier] Random delay: 3.7 seconds

**Comment by alainsch** [comment]
Since it is not working in postman and on de i-de site, I think it is a technical problem at I-DE

Outlook voor Android<https://aka.ms/AAb9ysg> downloaden
________________________________

**Comment by alejoar** [comment]
@alainsch did it eventually get fixed? I am experiencing this issue at the moment and my chart on home assistant looks flat for the past 5 days.

**Comment by alainsch** [comment]
Hello,

No. They did re-installed my account after several weeks, but in the mean time I decided to stop using this integration and bough my a Shelly EM with 2 x 50 A clamps.

It is not 100% correct as this doesn’t read the meter, but I have a better result because I now also have the numbers after the decimal point and I don’t have any dependency on iDE anymore for my statistics …

https://www.shelly.cloud/en/products/shop/em-120a/1xem-50a


[Image]
Alain Schellinck
+32 475 67 20 52


Van: Alejo Arias ***@***.***>
Verzonden: maandag 20 februari 2023 11:21
Aan: ldotlopez/ha-ideenergy ***@***.***>
CC: Alain Schellinck ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [ldotlopez/ha-ideenergy] I-de meter reading down or broken? (Issue #21)


@alainsch<https://github.com/alainsch> did it eventually get fixed? I am experiencing this issue at the moment and my chart on home assistant looks flat for the past 5 days.

**Comment by ldotlopez** [comment]
Hi,

I-de API for direct readings it's unstable

Additionally you need to apply for an "advanced user" in the i-de website
(for instant data), regular user only has access to historical data.

If you want to continue digging enable logging for this integration and
post relevant information please.

**Comment by alejoar** [comment]
> Hello, No. They did re-installed my account after several weeks, but in the mean time I decided to stop using this integration and bough my a Shelly EM with 2 x 50 A clamps. It is not 100% correct as this doesn’t read the meter, but I have a better result because I now also have the numbers after the decimal point and I don’t have any dependency on iDE anymore for my statistics … https://www.shelly.cloud/en/products/shop/em-120a/1xem-50a

Thank you! I will look into getting one of those once I overcome my fear of messing with the electrical panel. Did you install it yourself?

> Hi, I-de API for direct readings it's unstable Additionally you need to apply for an "advanced user" in the i-de website (for instant data), regular user only has access to historical data. If you want to continue digging enable logging for this integration and post relevant information please.

@ldotlopez I am an "advanced user" and everything was working fine until Friday when the graphs went flat.


![Screenshot from 2023-02-20 12-31-31](https://user-images.githubusercontent.com/6714697/220095199-9c1dcf41-51ee-4c14-aa3d-9d64862e56ff.png)


Trying to get a measurement from i-de also fails: I see this suspicious error flag with chrome dev tools, "permitirConextion", set to false (I don't know if that means anything).

![image](https://user-images.githubusercontent.com/6714697/220096387-a8dcc859-df27-4cfb-98ac-714ae0e29743.png)

I just enabled debug logging, I will let you know if there's anything to share.

**Comment by alainsch** [comment]
Hello,

Yes, I did install it myself, but I’m a trained electrician…

But it is very easy: just connect the clamp tot the shelly and 2 wires from the power. The clamp goes over one of the main wires.

But as always: electricity kills, so you better watch out. If in doubt, hire someone knowledgeable.

My experience is also that the API is very unstable. I had the “advanced user” setting on my account, but even then I couldn’t read my data anymore…

[Image]
Alain Schellinck
+32 475 67 20 52


Van: Alejo Arias ***@***.***>
Verzonden: maandag 20 februari 2023 12:44
Aan: ldotlopez/ha-ideenergy ***@***.***>
CC: Alain Schellinck ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [ldotlopez/ha-ideenergy] I-de meter reading down or broken? (Issue #21)


Hello, No. They did re-installed my account after several weeks, but in the mean time I decided to stop using this integration and bough my a Shelly EM with 2 x 50 A clamps. It is not 100% correct as this doesn’t read the meter, but I have a better result because I now also have the numbers after the decimal point and I don’t have any dependency on iDE anymore for my statistics … https://www.shelly.cloud/en/products/shop/em-120a/1xem-50a

Thank you! I will look into getting one of those once I overcome my fear of messing with the electrical panel. Did you install it yourself?

Hi, I-de API for direct readings it's unstable Additionally you need to apply for an "advanced user" in the i-de website (for instant data), regular user only has access to historical data. If you want to continue digging enable logging for this integration and post relevant information please.

@ldotlopez<https://github.com/ldotlopez> I am an "advanced user" and everything was working fine until Friday when the graphs went flat.

[Screenshot from 2023-02-20 12-31-31]<https://user-images.githubusercontent.com/6714697/220095199-9c1dcf41-51ee-4c14-aa3d-9d64862e56ff.png>

Trying to get a measurement from i-de also fails: I see this suspicious error flag with chrome dev tools, "permitirConextion", set to false (I don't know if that means anything).

[image]<https://user-images.githubusercontent.com/6714697/220096387-a8dcc859-df27-4cfb-98ac-714ae0e29743.png>

I just enabled debug logging, I will let you know if there's anything to share.

**Comment by alejoar** [comment]
@ldotlopez can you share what this endpoint looks like for you when you are logged in to i-de?

https://www.i-de.es/consumidores/rest/escenarioNew/validarComunicacionContador/

I don't have another account to check, but I suspect the "permitirConexion" value is the key here. For me it's `false` and it may be that it was deliberately disabled for my account.

**Comment by ldotlopez** [comment]
@alejoar 

I have the same **response** from i-de servers: `permitirConexion=false` however the ide integration is reporting some readings today for my contract.

You can try with the command line utility from the [ideenergy](https://github.com/ldotlopez/ideenergy/) module:
`python -m ideenergy -u USER -p PASS  --get-measure`


![image](https://user-images.githubusercontent.com/99769/220614141-1b67813f-dc27-42d8-9ae7-c096efec0f6c.png)

![image](https://user-images.githubusercontent.com/99769/220614587-c646db31-8006-4204-8fbb-4d3661a817d0.png)

**Comment by alejoar** [comment]
@ldotlopez I get a 403 :/
```
2023-02-22 13:29:20.960 INFO client <ideenergy.Client username=<redacted>, contract=None>: successful authentication
2023-02-22 13:29:20.961 DEBUG client Requesting data to the ICP, may take up to a minute.
Request failed: Invalid response: 403 - Forbidden
```

I am an "advanced" user so I don't know why this is happening. Do you have any clue?

**Comment by ldotlopez** [comment]
The command line tool is as reliable as the i-de API, that 403 means
nothing but a reading means that everything is OK. Look at my graph, I
didn't have a successful update in 5 hours. May be i-de is saturated or
some city equipment in your area is broken.

I don't have any other clues, try using historical data instead if it fits
your needs and keep the instant sensorr running just it case.

Sorry

**Comment by alejoar** [comment]
Gotcha, thank you.

**Comment by alejoar** [comment]
I think I "fixed" it.

I deleted my i-de account completely and created another one immediately after with the same email. I couldn't log in at first (it said user doesn't exist -- even though I got the "user activation/welcome" email) but 24 hours later I got a second "account activation/welcome" email and now I can log in.

I got a new reading (big one making up for the last few days). 

Hopefully it will keep working.

![image](https://user-images.githubusercontent.com/6714697/220866678-3d93ee57-272b-44f4-ae0f-ff2ce45059d4.png)
